### PR TITLE
chore(infrastructure): update dependencies

### PR DESCRIPTION
**Pull request by Goldziher**
This PR updates the dependencies. Additionally it handles skipping the beanie tests locally - since we do not install beanie except in CI.

### CODE REVIEW

**Comment by Goldziher** [comment]
> This is currently failing because it downgrades Pydantic to <2

this is what poetry 1.6 does, for some weird reason.

**Comment by Goldziher** [comment]
> > > This is currently failing because it downgrades Pydantic to <2
> > 
> > 
> > this is what poetry 1.6 does, for some weird reason.
> 
> Another poetry pain point..

ok, lets kill it

**Comment by Goldziher** [comment]
ok, this is ready to be merged. lets do the PDM and hatch stuff in a later scope. @provinzkraut - please approve and merge if you can.

**Review by provinzkraut — APPROVED** [review]

File: tests/unit/test_contrib/test_pydantic/test_beanie_integration.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from typing import Optional
+
+import pydantic
+import pytest
+
+from litestar.contrib.pydantic import PydanticDTO
+
+
+@pytest.mark.skipif(pydantic.VERSION.startswith("2"), reason="Beanie does not support pydantic 2 yet")
```
Comment (gsakkis) [inline tests/unit/test_contrib/test_pydantic/test_beanie_integration.py:9]: Beanie supports pydantic 2 in its latest version.

Comment (provinzkraut) [reply tests/unit/test_contrib/test_pydantic/test_beanie_integration.py:9]: We should test that then!

Comment (Goldziher) [reply tests/unit/test_contrib/test_pydantic/test_beanie_integration.py:9]: PR welcome

**Pull request merged**